sdk/storage/azure-storage-file/azure/storage/file/_parser.py

```diff
@@ -0,0 +1,48 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from datetime import datetime, timedelta
+
+_ERROR_TOO_MANY_FILE_PERMISSIONS = 'file_permission and file_permission_key should not be set at the same time'
+_FILE_PERMISSION_TOO_LONG = 'Size of file_permission is too large. file_permission should be <=8KB, else' \
+                            'please use file_permission_key'
+
+
+def _get_file_permission(file_permission, file_permission_key, default_permission):
+    # if file_permission and file_permission_key are both empty, then use the default_permission
+    # value as file permission, file_permission size should be <= 8KB, else file permission_key should be used
+    empty_file_permission = not file_permission
+    empty_file_permission_key = not file_permission_key
+    file_permission_size_too_big = False if file_permission is None \
+        else len(str(file_permission).encode('utf-8')) > 8 * 1024
+
+    if file_permission_size_too_big:
+        raise ValueError(_FILE_PERMISSION_TOO_LONG)
+
+    if empty_file_permission:
+        if empty_file_permission_key:
+            return default_permission
+
+        return None
```

Comment:
just curious: if file_permission is empty and file_permission_key is not, shouldn't this return the key?

Reply:
No - if you check the function that call this one - file_permission_key is used independently. I checked as I had the same question ;)

Reply:
Thanks for clarifying. 😄